oak-lucene/src/test/java/org/apache/jackrabbit/oak/composite/blueGreen/IndexUtils.java

```diff
@@ -147,4 +155,38 @@ public static void checkLibsIsReadOnly(Persistence p) throws RepositoryException
         }
     }
 
+    public static boolean isIndexDisabledAndHiddenNodesDeleted(NodeStore store, String path) throws CommitFailedException {
+        NodeState nodeState = NodeStateUtils.getNode(store.getRoot(), path);
+        if (!nodeState.getProperty("type").getValue(Type.STRING).equals("disabled")) {
```

Comment:
Comment: getProperty could return null, but as this is just a test, I think it's fine and we don't need to check for null. But in production code, we would need to verify. In the test case, I actually prefer shorter / simpler code, even if it's slightly dangerous. The only problem is when someone moves the code to /src/main. What about adding a comment

// could throw NPE, but we don't care as it's a test

So this won't be forgotten when the code is moved